piccolo_cursor_pagination/pagination.py

```diff
@@ -0,0 +1,153 @@
+                            == last_row["id"]
+                        ):
+                            headers["cursor"] = ""
+                    except IndexError:
+                        # no more results
+                        headers["cursor"] = ""
+            except KeyError:
+                query = query.where(
+                    table._meta.primary_key >= int(decoded_cursor)
+                ).limit(self.page_size + 1)
+                rows = query.run_sync()
+                # set new value to next_cursor
+                next_cursor = self.encode_cursor(str(rows[-1]["id"]))
+                headers["cursor"] = next_cursor
+
+        else:
+            # query where limit is equal to page_size plus
+            # one in DESC order
+            query = table.select().order_by(
+                table._meta.primary_key, ascending=False
+            )
+            query = query.limit(self.page_size + 1)
+
+            # decoded query params cursor
+            decoded_cursor = self.decode_cursor(self.cursor, table)
+
+            try:
+                # preform operation if previous in query params
+                previous = request.query_params["__previous"]
+                if previous:
+                    query = (
+                        query.where(
+                            table._meta.primary_key > int(decoded_cursor)
+                        )
+                        .order_by(table._meta.primary_key, ascending=True)
+                        .limit(self.page_size)
+                    )
+                    rows = query.run_sync()
+                    try:
+                        # set new value to next_cursor
+                        next_cursor = self.encode_cursor(str(rows[-1]["id"]))
+                        headers["cursor"] = next_cursor
+                        # return empty cursor if no more results
+                        last_row = (
+                            table.select()
+                            .limit(1)
+                            .order_by(table._meta.primary_key, ascending=False)
+                            .first()
+                            .run_sync()
+                        )
+                        if (
+                            self.decode_cursor(next_cursor, table)
+                            == last_row["id"]
+                        ):
+                            headers["cursor"] = ""
+                    except IndexError:
+                        # no more results
+                        headers["cursor"] = ""
+            except KeyError:
+                query = query.where(
+                    table._meta.primary_key <= int(decoded_cursor)
+                ).limit(self.page_size + 1)
+                rows = query.run_sync()
+                # set new value to next_cursor
+                next_cursor = self.encode_cursor(str(rows[-1]["id"]))
+                headers["cursor"] = next_cursor
+
+        query = query.limit(self.page_size)
+        return query, headers
```

Comment:
Generally I think returning a dataclass is nicer than returning a tuple - up to you.

Reply:
If you don’t mind, we should leave this as it is because tuples is easiest way to return multiple values.

Reply:
That's fine 👍